packages/runtime/container-runtime/src/containerRuntime.ts

```diff
@@ -1038,6 +1049,13 @@ export class ContainerRuntime extends TypedEventEmitter<IContainerRuntimeEvents>
             this.validateSummaryHeuristicConfiguration(this.summaryConfiguration);
         }
 
+        this.enableOpReentryCheck = (runtimeOptions.enableOpReentryCheck === true
+            // If compression is enabled, we need to disallow op reentry as it is required that
+            // ops within the same batch have the same reference sequence number.
+            || runtimeOptions.compressionOptions.minimumBatchSizeInBytes !== Number.POSITIVE_INFINITY)
+            // Allow for a break-glass config to override the options
+            && this.mc.config.getBoolean("Fluid.ContainerRuntime.DisableOpReentryCheck") !== true;
```

Comment:
Do we need runtimeOptions.enableOpReentryCheck? I.e. should we have fewer ways to enable same thing? Feature gate should be enough - it should be easy (if client wants so) to programmatically always set the settings (and overwrite control tower in code).

Reply:
I think we do. There's one way to enable this, via runtime options. The feature gate is a break-glass way of disabling it via control tower.